client/cpp.cpp

```diff
@@ -116,36 +116,7 @@ pid_t call_cpp(CompileJob &job, int fdwrite, int fdread)
         appendList(flags, job.restFlags());
 
         for (list<string>::iterator it = flags.begin(); it != flags.end();) {
-            /* This has a duplicate meaning. it can either include a file
-               for preprocessing or a precompiled header. decide which one.  */
-            if ((*it) == "-include") {
-                ++it;
-
-                if (it != flags.end()) {
-                    std::string p = (*it);
-
-                    if (access(p.c_str(), R_OK) < 0 && access((p + ".gch").c_str(), R_OK) == 0) {
-                        // PCH is useless for preprocessing, ignore the flag.
-                        list<string>::iterator o = --it;
-                        ++it;
-                        flags.erase(o);
-                        o = it++;
-                        flags.erase(o);
-                    }
-                }
-            } else if ((*it) == "-include-pch") {
-                list<string>::iterator o = it;
-                ++it;
-                if (it != flags.end()) {
-                    std::string p = (*it);
-                    if (access(p.c_str(), R_OK) == 0) {
-                        // PCH is useless for preprocessing (and probably slows things down), ignore the flag.
-                        flags.erase(o);
-                        o = it++;
-                        flags.erase(o);
-                    }
-                }
```

Comment:
What trouble exactly is this part supposed to cause? The flags are removed for a reason, PCH files are not sent to the remote node, so if the flags are kept, the remote compilation may fail.

Reply:
In my environment, that code made building fail.
I'm using `Prefix Header` option on Xcode. 
This option help for not including common headers which is always included.
For example, As I set `prefixHeader.pch` as value of `Prefix Header` option, I don't need to include stdarg or vector or any other in my sourcecode.
```cpp
// contents of prefixHeader.pch
#include <stdarg.h>
#include <stdlib.h>
#include <string.h>
#include <limits.h>
#include <float.h>
#include <math.h>
#include <stdio.h>
#include <time.h>
#include <stdint.h>
#include <assert.h>
#include <sys/time.h>
#include <sys/select.h>
#include <OpenGLES/ES2/gl.h>
#include <OpenGLES/ES2/glext.h>

#ifdef __OBJC__
    #import <Foundation/Foundation.h>
    #import <UIKit/UIKit.h>
#endif

#ifdef __cplusplus
    #include <vector>
    #include <string>
    #include <map>
    #include <list>
    #include <memory>
    #include <functional>
    #include <thread>
    #include <mutex>
    #include <unordered_map>
    #include <list>
    #include <algorithm>
    #include <cstdint>
    #include <stack>
#endif
```
But code that I removed make "-include prefixHeader.pch" skipped during local compilation, So build is failed because that source has not included stdio or vector or etc...

After some investigation, I found you removed the smilar code(#475 ).
I have thought the comment in the #475 can be also applied to the code I removed.
If it must not be removed, I have to find other way to support `Prefix Header` option.

Reply:
Bleows are documentations in the Xcode.

> 
> Precompile Prefix Header (GCC_PRECOMPILE_PREFIX_HEADER)
> Generates a precompiled header for the prefix header, which should reduce overall build times.
> 
> Precompiling the prefix header will be most effective if the contents of the prefix header or any file it includes change rarely. If the contents of the prefix header or any file it includes change frequently, there may be a negative impact to overall build time.
> 
> Prefix Header (GCC_PREFIX_HEADER)
> Implicitly include the named header. The path given should either be a project relative path or an absolute path.
> 
> Preprocessor Macros (GCC_PREPROCESSOR_DEFINITIONS)
> Space-separated list of preprocessor macros of the form foo or foo=bar.
>